src/panels/lovelace/cards/hui-area-card.ts

```diff
@@ -217,24 +266,25 @@ export class HuiAreaCard
       return undefined;
     }
     let uom;
-    const values = entities.filter((entity) => {
-      if (!isNumericState(entity) || isNaN(Number(entity.state))) {
-        return false;
-      }
-      if (!uom) {
-        uom = entity.attributes.unit_of_measurement;
-        return true;
-      }
-      return entity.attributes.unit_of_measurement === uom;
-    });
+    const values = entities
+      .filter((entity) => {
+        if (!isNumericState(entity) || isNaN(Number(entity.state))) {
+          return false;
+        }
+        if (!uom) {
+          uom = entity.attributes.unit_of_measurement;
+          return true;
+        }
+        return entity.attributes.unit_of_measurement === uom;
+      })
+      .map((entity) => Number(entity.state));
     if (!values.length) {
       return undefined;
     }
-    const sum = values.reduce(
-      (total, entity) => total + Number(entity.state),
-      0
-    );
-    return `${formatNumber(sum / values.length, this.hass!.locale, {
+    const displayValue = deviceClass
+      ? this._deviceClasses[deviceClass].default_display_function(values)
+      : "";
+    return `${formatNumber(displayValue, this.hass!.locale, {
```

Comment:
This change specifically addresses the issue at hand (all Area card numeric sensor values were averaged). All other changes are ancillary to these lines.